platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/StateSigner.java

```diff
@@ -59,7 +61,7 @@ public StateSigner(@NonNull final PlatformSigner signer, @NonNull final Platform
      * @param reservedSignedState the state to sign
      * @return a {@link StateSignaturePayload} containing the signature, or null if the state should not be signed
      */
-    public @Nullable StateSignaturePayload signState(@NonNull final ReservedSignedState reservedSignedState) {
+    public @Nullable ConsensusTransactionImpl signState(@NonNull final ReservedSignedState reservedSignedState) {
```

Comment:
The problem is that `ConsensusTransactionImpl` is not freely interchangeable with `ConsensusTransaction`, a bunch of APIs expect the `impl` type, and I'd have to cast in order to make things work. Attempting to avoid going down the "fixing consensus interfaces" rabbit hole as a part of this PR (although it's something that desperately needs doing).